psi4/src/psi4/libmints/matrix.h

```diff
@@ -1044,20 +1058,50 @@ class PSI_API Matrix : public std::enable_shared_from_this<Matrix> {
      * @param theta - the angle (in radians) about which to rotate
      */
     void rotate_columns(int h, int i, int j, double theta);
-    friend class Vector;
-};
 
-namespace detail {
-/*!
- * allocate a block matrix -- analogous to libciomr's block_matrix
- */
-double** matrix(int nrow, int ncol);
+    PSI_DEPRECATED(
+        "Using `Matrix::matrix` instead of `detail::matrix` is deprecated, and in 1.4 it will "
+        "stop working")
+    static double** matrix(int nrow, int ncol) { return detail::matrix(nrow, ncol); }
+
+    PSI_DEPRECATED(
+        "Using `Matrix::free` instead of `detail::free` is deprecated, and in 1.4 it will "
+        "stop working")
+    static void free(double** Block) { detail::free(Block); }
+
+    PSI_DEPRECATED(
+        "Using `Matrix::create` instead of `create` is deprecated, and in 1.4 it will "
+        "stop working")
+    static SharedMatrix create(const std::string& name, const Dimension& rows, const Dimension& cols) {
+        return create(name, rows, cols);
+    }
 
-/*!
- * free a (block) matrix -- analogous to libciomr's free_block
- */
-void free(double** Block);
-}  // namespace detail
+    PSI_DEPRECATED(
+        "Using `Matrix::horzcat` instead of `horzcat` is deprecated, and in 1.4 it will "
```

Comment:
should this (and some others) have `detail::` in the message?

Reply:
No, because these have been moved to the `psi` namespace.